### DISCUSSION TITLE
when doing the client-side navigation, is there a way for client-side to dynamically disable `getServersideProps` to prevent duplicated requests?

### DISCUSSION THREAD

**Discussion by RaynorLeee** [vercel/next.js#68861] [Help]
### Summary

see demo:
https://codesandbox.io/p/devbox/mc-demo-9qm6kd
and goto: 
https://9qm6kd-3000.csb.app/play

I have 2 pages, a `game lobby`(https://9qm6kd-3000.csb.app/play) page and a `play game`(https://9qm6kd-3000.csb.app/play/game_1234) page, which is dynamic page route(`pages/play/[[...games]].tsx`)

upon user entering this page, we need to fetch game list, which was done by using `getServerSideProps`. this list could be huge, here I made a 10,000-member list

**the problem is**: when user switching back and forth between `lobby` page and `game-play` page, the `getServerSideProps` gets called EVERY SINGLE TIME! we already have the `gameList` data on the FE component's cache, which requesting it again and again is a waste of traffic and time. 

So my question is: is there a way for client-side, at least to let server-side know the data is already exist in the client-side, or even better, dynamically disable this page's `getServerSideProps` based on what we have on the client-side?

Thanks! 

### Example

https://codesandbox.io/p/devbox/mc-demo-9qm6kd

**Comment by icyJoseph** [comment]
Have you tried adding the shallow prop to the next/link's used to navigate between these?

**Reply by RaynorLeee** [reply]
thanks! it do works in the demo, but not in my prod project...weird.
